### DISCUSSION TITLE
BUG: `definePage` causes jsx error in version 0.10.0

### DISCUSSION THREAD

**Discussion by A-kirami** [posva/unplugin-vue-router#429] [Questions and Help] [answered]
**Description:**
After updating to version 0.10.0, I encountered the following error when using `definePage`:
```
12:36:36 [vite] Internal server error: Unexpected token, expected "}" (3:10)
  Plugin: unplugin-vue-jsx-directive
  File: C:/Users/.../src/views/index.vue?definePage&vue&lang.tsx:3:10
  1  |  <script setup lang="ts">
  2  |  definePage({
  3  |    redirect: '/',
     |              ^
  4  |  })
  5  |  </script>
```

I have tested this issue and confirmed that it does not occur in version 0.9.1.

**Possible Cause:**
I suspect that the issue might be related to this commit: https://github.com/posva/unplugin-vue-router/commit/4fbff23666bb3fcf9ad7fcbc0d6ceda9d2c45d7f

Please let me know if any additional information is needed. Thank you!

**Comment by posva** [comment, marked as answer]
The error comes from unplugin-vue-jsx-directive not from here. It seems to be picking up the `.vue` file. Maybe this is a feature in the playing, in which case I might be able to add some other query. As long as the esbuild plugins picks it up, it's fine. But the reason I added jsx is to allow jsx in the definePage. Maybe that was a mistake and adding jsx there isn't needed?
What is weid though is that the other plugin fails

**Comment by 0asys** [comment]
I have the same error:

```
[plugin:vite-plugin-vue-inspector] unknown: Unexpected token, expected "}" (5:6)
  8 | })
src/ui/pages/settings.vue:5:6
3  |  
4  |  definePage({
5  |    meta: {
   |        ^
6  |      requiresAuth: true
7  |    },
```

originating from `vite-plugin-vue-inspector` ?

**Reply by 0asys** [reply]
the culprit was: `vite-plugin-vue-devtools`

**Reply by starredev** [reply]
> the culprit was: `vite-plugin-vue-devtools`

you are correct. @posva after 10.0 it seems it has issues in combination with vue devtools.

**Reply by TaKeOvEr-DeV** [reply]
Change  vite-plugin-vue-devtools to the end, in the plugin array. @bpereto @posva 
It works correctly.
 @bpereto @posva 
![Captura de pantalla 2024-11-28 161051](https://github.com/user-attachments/assets/7e56ce89-217e-4896-b792-b38bd9311f09)

**Reply by CarlosIvanHuerta** [reply]
thank you @TaKeOvEr-DeV 
That's great! Moving the `DevTools()` plugin to the **end of the plugin list** is a very logical solution, as plugins in Vite are processed in the order they appear in the list. By placing it at the end:
1. **You avoid conflicts**: Other plugins, such as `vue`, or any internal plugins, run first without interference from the . `vue-router``vite-plugin-vue-devtools` plugin.
2. **It inspects correctly**: DevTools can work with the final processing results of other plugins (e.g., generated routes, rendered components, etc.).
